### PR TITLE
docs: add contribution guide

### LINKED ISSUE #188: Provide a 'contribution-guide' for this project
I'm currently contributing for the first time to this project, and as a non-Powershell expert, it would be nice to have some guidance available in the form of documentation.

This guide could describe what steps you need to take for certain topics:

- How to create a new module
- How to create a new script (create a ps1 file, make the required modifications in the psd1 and psm1 files, .... )
- How to create a unit test
- How to run the unit-tests  (I'm currently struggling on this step for instance, when running `Invoke-Pester -Path <file>`, I'm always faced with the error 'Module <modulename> is not a Script module>'.

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add contribution guide to the repo to explain and guide new contributers.

Closes #188

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# Contributing to Arcus.Scripting
+🎉 First off, thanks for taking the time to contribute! We're really glad you're reading this. 🎉
+
+The following set of guidelines will help you contributing to this Arcus repository. Different than other repositories, it's written in PowerShell which can be an extra hurtle to take for some.
```
Comment (fgheysels) [inline CONTRIBUTING.md:4]:
```suggestion
The following set of guidelines will help you to contribute to this Arcus repository. The difference with other Arcus repositories, is that `Arcus.Scripting` is written in PowerShell which can be an extra hurdle to take for some.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+But, no worries, if you forget something or have problems with some of these steps, we'll happy to help! In those cases: create a PullRequest on this repository so we can talk about it.
```
Comment (fgheysels) [inline CONTRIBUTING.md:6]:
```suggestion
But, no worries, if you forget something or have problems with some of these steps, we're happy to help! In those cases: create a PullRequest on this repository so we can talk about it.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+Each of our modules consists of the following structure:
+```
+Arcus.Scripting.ModuleName\
+- Scripts\
+- Arcus.Scripting.ModuleName.psm1
+- Arcus.Scripting.ModuleName.psd1
+```
+
+### `.psd1` file
+This file is the module manifest file of the PowerShell module. This contains metadata information like the module version, required dependency modules, module description, etc.
```
Comment (fgheysels) [inline CONTRIBUTING.md:20]:
```suggestion
This file is the module manifest file for the PowerShell module. This manifest contains metadata information such as: the module version, required dependency modules, module description, etc...
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+### `.psm1` file
+This file is the script module file. This contains the actual exposed functionality that will be available when the PowerShell module gets installed.
```
Comment (fgheysels) [inline CONTRIBUTING.md:23]:
```suggestion
This file is the script module file. This file describes the functionality that is exposed and that will be available when the PowerShell module is installed.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+This folder contains all the PowerShell scripts in the module. As you will see, each of these scripts will have corresponding links with the `.psm1` file to expose it.
+
+For more information on PowerShell modules, see [the official Microsoft docs](https://docs.microsoft.com/en-us/powershell/scripting/developer/module/understanding-a-windows-powershell-module).
+
+## How to add a new script in an existing module
+A new script to an existing PowerShell module means that the script will be available as a module function.
+These are the steps you need to take:
+
+1. Add your PowerShell `.ps1` script file to the `Scripts\` folder of the PowerShell module you want to extend.
+2. Expose your script in the `.psm1` script module. This is normally something in the line of this.
```
Comment (fgheysels) [inline CONTRIBUTING.md:35]:
```suggestion
2. Expose your script in the `.psm1` script module. This normally looks like this:
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+    . $PSScriptRoot\Scripts\My-AzCustomScript.ps1 -MyParameter $MyParameter
+}
+
+Export-ModuleMember -Function My-AzCustomFunction
+```
+
+3. Add your function name to the `FunctionsToExport` in the `.psd1` file: `FunctionsToExport = @('My-AzCustomFunction')`
+
+Voila! Now your script will be part as a module function the next time the PowerShell module will be installed.
```
Comment (fgheysels) [inline CONTRIBUTING.md:61]:
```suggestion
Voila! Now your script will be available as a module function the next time the PowerShell module is installed.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+                # Act
+
+                # Assert
+            }
+        }
+    }
+}
+```
+
+When you want to tests these locally, we recommand to provide a valid `ModuleVersion` in the `.psd1` file, as it's required for loading the module in Pester.
```
Comment (fgheysels) [inline CONTRIBUTING.md:99]:
```suggestion
When you want to run these tests locally, we recommend to provide a valid `ModuleVersion` in the `.psd1` file, as it's required for loading the module in Pester.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+```powershell
+Invoke-Pester .\src\Arcus.Scripting.Unit.Tests\Arcus.Scripting.YourModule.tests.ps1
+```
+
+## How to add a new module
+We use the [PowerShell Tools for Visual Studio](https://ironmansoftware.com/powershell-pro-tools) to manage our project. When creating a new PowerShell module, this corresponds with creating a new PowerShell project to our code solution. Make sure that you name the module in the format `Arcus.Scripting.*`.
```
Comment (fgheysels) [inline CONTRIBUTING.md:106]:
```suggestion
We use the [PowerShell Tools for Visual Studio](https://ironmansoftware.com/powershell-pro-tools) to manage our project. When creating a new PowerShell module, this corresponds with adding a new PowerShell project to our code solution. Make sure that you name the module in the format `Arcus.Scripting.*`.
```

I'd say:
> ... creating a new ....  in our code solution
or 
> ... adding a new ... to our code solution

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        ProjectUri = 'https://github.com/arcus-azure/arcus.scripting'
+        IconUri = 'https://raw.githubusercontent.com/arcus-azure/arcus/master/media/arcus.png'
+        ReleaseNotes = 'https://github.com/arcus-azure/arcus.scripting/releases/tag/v#{Package.Version}#'
+    }
+  }
+  ```
+
+    > Note that any PowerShell dependencies are to be added to the `RequiredModules` in the form of `RequiredModules = @{ @{ ModuleName = 'YourDependency'; ModuleVersion = '1.2.3' } }`
+
+These is all the important parts. After you updated this file, you're good to go!
```
Comment (fgheysels) [inline CONTRIBUTING.md:128]:
```suggestion
These are all the important parts. After you updated this file, you're good to go!
```

**Review by fgheysels — COMMENTED** [review]
Thanks for this! This will be very helpful for 1st-time (and  other) contributors, I'm talking from experience :)

Just added some suggestions.

**Comment by stijnmoreels** [comment]
> Thanks for this! This will be very helpful for 1st-time (and other) contributors, I'm talking from experience :)
> 
> Just added some suggestions.

Great! Thanks for the valuable feedback here.

**Comment by mbraekman** [comment]
Great addition! Thanks a lot!

Just 2 remarks:
- Can we include a section about how/where to update the documentation?
- Can we include a link to this page from the "index"-page of the documentation? Makes it easy to find.

**Comment by stijnmoreels** [comment]
> * Can we include a link to this page from the "index"-page of the documentation? Makes it easy to find.

Hmm, the problem here is that GitHub requires us to place the `CONTRIBUTING.md` file in the root directory, while our documentation files are build in a `./docs` sub-directory. Not sure if the documentation site will be happy to reference this.

**Comment by stijnmoreels** [comment]
> * Can we include a section about how/where to update the documentation?

@Mcalewaert  was so friendly to include a 'how to' page for documentation on every Arcus repo:
https://github.com/arcus-azure/arcus.scripting/blob/master/docs/README.md

Maybe just reference this?

**Comment by mbraekman** [comment]
> Hmm, the problem here is that GitHub requires us to place the `CONTRIBUTING.md` file in the root directory, while our documentation files are build in a `./docs` sub-directory. Not sure if the documentation site will be happy to reference this.

Aha, wasn't aware of that.
Then perhaps add it on the README.md at least to make it a bit easier to find.

**Comment by mbraekman** [comment]
> > * Can we include a section about how/where to update the documentation?
> 
> @Mcalewaert was so friendly to include a 'how to' page for documentation on every Arcus repo:
> https://github.com/arcus-azure/arcus.scripting/blob/master/docs/README.md
> 
> Maybe just reference this?

Indeed, let's add a reference to that page :)

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -5,6 +5,13 @@ The following set of guidelines will help you in contributing to this Arcus repo
 
 But, no worries, if you forget something or have problems with some of these steps, we're happy to help! In those cases: create a PullRequest on this repository so we can talk about it.
 
+This contribution guide has the following contents:
+- [PowerShell modules](#powershell-modules)
+- [How to add a new script in an existing module](#how-to-add-a-new-script-to-an-existing-module)
```
Comment (mbraekman) [inline CONTRIBUTING.md:10]: The link currently doesn't work, due to the difference in "**in** an existing" vs "**to** an existing".
Should be "to an existing module".
Can you adjust this in the link here and the header further down the page?

```suggestion
- [How to add a new script to an existing module](#how-to-add-a-new-script-to-an-existing-module)
```

**Review by mbraekman — APPROVED** [review]
One small thing, after that: ship it!

File: CONTRIBUTING.md (hunk 10 above)
Comment (stijnmoreels) [reply CONTRIBUTING.md]: Aha, thanks! Missed that one. Fixed it.

**Pull request merged**